### PR TITLE
Sgds 505 mainnav

### PR DESCRIPTION
**Pull request by rav3**
### css custom variables for .sgds.navbar
```
  --mainnav-background-color: white;
  --mainnav-padding-x: 1rem;
  --mainnav-padding-y: 0;
  --mainnav-mobile-padding-x: 0.5rem;
  --mainnav-mobile-padding-y: 0;
  --mainnav-borderBottom-width: 1px;
  --mainnav-borderBottom-color: #98a2b3;
```

### CODE REVIEW

File: src/base/dropdown-element.ts [hunk 1]
```diff
@@ -24,43 +24,71 @@ export type DropdownButtonVariant =
...
   @property({ type: String, reflect: true })
   drop: DropDirection = "down";
+
+  /** Additional configuration to pass to Popper.js. See https://popper.js.org/ for config opts */
   @property({ type: Object })
   popperOpts = {};
+
+  /** Forwards value to id attribute of toggle button of Dropdown. An unique id generated by default */
   @property({ type: String })
   togglerId = genId("dropdown", "button");
```
Comment (clukhei) [inline src/base/dropdown-element.ts:60]: can help me add typing  "string" here. I think the custom element analzyer cannot detect genId returns a string 

`togglerId : string = genId("dropdown", "button")`

**Review by clukhei — APPROVED** [review]
Hello, will you be fixing the mainnav dropdown link color to black?

All else is looking good !

File: src/components/Mainnav/sgds-mainnav-item.ts [hunk 2]
```diff
@@ -3,11 +3,15 @@ import styles from "./sgds-mainnav-item.scss";
 import LinkElement from "../../base/link-element";
 
 /**
- * @slot - Label for the anchor tag.
+ * @slot default - slot for SgdsMainnavItem element.
+ *
+ * @cssproperty --mainnav-item-theme-color - Hover and active color for mainnav items.
+ * @cssproperty --mainnav-item-color - Text color of nav item.
+ * @cssproperty --mainnav-item-borderBottom-width - border bottom width for hover and active state for nav item
```
[the remark this answers is not in the document]
Comment (rav3) [reply src/components/Mainnav/sgds-mainnav-item.ts:10]: > Hello, will you be fixing the mainnav dropdown link color to black?
> 
> All else is looking good !

Oh yes!! i missed that portion. Will do


> OMG @cssproperty also work on storybook table lehh. COol

Indeed!! haha

File: src/base/dropdown-element.ts (hunk 1 above)
Comment (rav3) [reply src/base/dropdown-element.ts]: Resolved in [40d2e42](https://github.com/GovTechSG/sgds-web-component/pull/54/commits/40d2e42a0eb72c750567fb021865f04ca94c517c)

File: src/components/Mainnav/sgds-mainnav-item.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rav3) [reply src/components/Mainnav/sgds-mainnav-item.ts:10]: Added @cssproperty example in the CODE_COVENTION 
[d1b8032](https://github.com/GovTechSG/sgds-web-component/pull/54/commits/d1b803207e88c78c90521506ea613770cdd67dbc)

**Pull request merged**